### ISSUE TITLE
A question about the use of "qa_id" of each dataset item

### ISSUE THREAD

**Issue by qwqwq1445** [jihoontack/MAC#1]
Thanks for your great work and I am a new guy in this research area. I try to follow your code and one question confuses me. In the common/dataset.py, we just concat the question and answer pair to form the input_ids and attention_masks.
![image](https://github.com/jihoontack/MAC/assets/65895217/abec497f-a31c-4aaf-b4c3-027c36eea8be)
And in the models/amortized_enddec.py, we input the "qa_ids" and "qa_attention" into base_lm for the "qa_loss". 
![image](https://github.com/jihoontack/MAC/assets/65895217/1c5c4c3a-859b-4dc0-bad1-da575245c909)
I wonder if this will cause information leak to the llm, since we already input the answer with the question to it? As I mentioned I am new in this area and not familiar with the whole online adaptation pipeline. Would you kindly please give me some ideas? Looking forward to your reply.

**Comment by jihoontack** [comment]
Hi, thank you for your interest in our paper! 

This is a typical way to define QA loss in the literature, and it is unrelated to online adaptation. Specifically, we use teacher forcing, which is commonly used for autoregressive modeling (using the ground truth previous tokens to predict the next token). In the QA setup, it is using the previous answer token (and the question tokens) to predict the next answer token. 

Also, note that this is used for the training stage, hence, there is no leakage in the online adaptation stage (we do not use the answer when adapting and generating the answer).
- https://github.com/jihoontack/MAC/blob/main/evals/qa_utils.py#L91
- This part generates the answer where we do not use the answer.

We are still refactoring and polishing the code, so if you have any bugs or issues, please let us know! 

Thank you again for your interest.

Best,
Jihoon

**Comment by qwqwq1445** [comment]
Thanks for your kind help. I have another question: what is the "lift_ratio" in the file config?
